app/services/stripe/cancel_subscription_service.rb

```diff
@@ -0,0 +1,19 @@
+module Stripe
+  class CancelSubscriptionService
+    def initialize(subscription)
+      @subscription = subscription
+    end
+
+    def call
+      return false unless @subscription.stripe_subscription_id.present?
+
+      begin
+        @subscription.cancel
+        @subscription.cancel_stripe_subscription
```

Comment:
what if stripe subscription fails but your database subscription will be cancelled ? both need to be at same state

Reply:
it will only delete spree subscription if we delete stripe subscription in my new code

Reply:
yes this is what i am seeing, if stripe subscription deletion fails then our database subscripton also should not succeed. this can lead to inconsistent data state

Reply:
in your sequence, stripe subscription will be cancelled and then you db will try to cancel, if your db fails to update due to some reason, does your wrapper`ActiveRecord::Base.transaction` help?